pkg/services/ngalert/state/manager.go

```diff
@@ -169,15 +169,37 @@ func (st *Manager) ResetStateByRuleUID(ctx context.Context, ruleKey ngModels.Ale
 // ProcessEvalResults updates the current states that belong to a rule with the evaluation results.
 // if extraLabels is not empty, those labels will be added to every state. The extraLabels take precedence over rule labels and result labels
 func (st *Manager) ProcessEvalResults(ctx context.Context, evaluatedAt time.Time, alertRule *ngModels.AlertRule, results eval.Results, extraLabels data.Labels) []*State {
+	var (
+		states        []StateTransition
+		optionalImage *ngModels.Image
+	)
+
 	logger := st.log.FromContext(ctx)
 	logger.Debug("State manager processing evaluation results", "resultCount", len(results))
-	var states []StateTransition
 
 	for _, result := range results {
```

Comment:
There is a bug here where either if the alert rule is deleted, or a firing series stops returning data while another series is `Normal`, then no image will be taken. However, an image should be present in those cases.